File: ScoreFrame/ScoreFrame/Views/Common/TimeFormatting.swift
import Foundation
import CoreMedia

struct TimeFormatting {
    static func format(seconds: TimeInterval) -> String {
        let totalSeconds = Int(max(0, seconds))
        let minutes = totalSeconds / 60
        let secs = totalSeconds % 60
        return String(format: "%02d:%02d", minutes, secs)
    }

    static func format(cmTime: CMTime) -> String {
        guard cmTime.isValid, !cmTime.isIndefinite else { return "00:00" }
        return format(seconds: cmTime.seconds)
    }
}

File: OverScore-Android/app/src/main/java/com/overscore/util/TimeFormatting.kt
package com.overscore.util

import kotlin.math.max

object TimeFormatting {
    fun format(seconds: Double): String {
        val totalSeconds = max(0.0, seconds).toInt()
        val minutes = totalSeconds / 60
        val secs = totalSeconds % 60
        return "%02d:%02d".format(minutes, secs)
    }

    fun formatMillis(millis: Long): String {
        if (millis <= 0) return "00:00"
        return format(millis / 1000.0)
    }
}
